app/controllers/api/v2/ansible_roles_controller.rb

```diff
@@ -32,28 +33,46 @@ def destroy
         process_response @ansible_role.destroy
       end
 
-      api :PUT, '/ansible_roles/import', N_('Import Ansible roles')
+      api :PUT, '/ansible_roles/import', N_('DEPRECATED: Import Ansible roles'), deprecated: true
       param :proxy_id, :identifier, :required => true, :desc => N_('Smart Proxy to import from')
-      param :role_names, Array, N_('Ansible role names to import')
+      param :role_names, Array, N_('Ansible role names to imported')
       def import
-        @imported = @importer.import!(role_names)
+        Foreman::Deprecation.api_deprecation_warning(_('Use /ansible_roles/sync to sync role with ansible smart proxy'))
+        new_roles = @roles_importer.import_role_names['new']
+        if role_names.present?
+          new_roles.select! do |role|
+            role_names.include?(role.name)
+          end
+        end
+        new_roles.map(&:save)
+        @imported = new_roles
+      end
+
+      api :PUT, '/ansible_roles/sync', N_('DEPRECATED: Sync Ansible roles')
```

Comment:
Am I missing something? You've deprecated `/import` to use `/sync` and this is also deprecated?